### PR TITLE
Add system registers, remove invalid registers from intrinsics

### LINKED ISSUE #12: RDMSR/WRMSR reference invalid registers
Binja generates a 2-operand intrinsic for RDMSR, and a 3-operand intrinsic for WRMSR. The extra operand in both cases is an "invalid" register. Looks like a problem with the code generator. `iform-type-dump.txt` indicates that RDMSR reads "XED_REG_MSRS"... I'm guessing that means loosely "this thing reads some MSR(s)", and `parse-iform-types.py` doesn't special-case that.

### PR DESCRIPTION
**Pull request by nshp**
Fixes Vector35/arch-x86#12, Vector35/binaryninja-api#4033 and Vector35/binaryninja-api#4032

Coreboot is a nice big test case for x86 system code fwiw. It's pretty easy to build, and you can open `build/cbfs/fallback/bootblock.raw.elf` to get symbols and whatnot.

### CODE REVIEW

File: arch_x86.cpp [hunk 1]
```diff
@@ -1927,7 +1927,42 @@ BNEndianness X86CommonArchitecture::GetEndianness() const
...
+		XED_REG_GS,
+		XED_REG_FSBASE,
+		XED_REG_GSBASE,
+		XED_REG_FLAGS,
+		XED_REG_EFLAGS,
+		XED_REG_RFLAGS
+	};
+}
+
+vector<uint32_t> X86CommonArchitecture::GetSystemRegisters()
```
Comment (jeffli678) [inline arch_x86.cpp:1945]: This change is good in general. However, there are other registers that can be considered system registers, e.g., DR0-DR7. I am not super familiar with the effect of GetSystemRegisters(), which I will discuss with the team and get back to you. It is likely some changes are needed here eventually

**Review by jeffli678 — COMMENTED** [review]
Thanks for the PR! I will discuss with the team and offer more comments later.

File: arch_x86.cpp (hunk 1 above)
Comment (nshp) [reply arch_x86.cpp:1945]: I agree this list is not complete, it's just a step in the right direction.

As I understand it, system registers should be for anything where a read won't necessarily match what you write (if you even can write). Like control registers, where you write a command and read back a status. This designation should be used in binja to exclude these registers from both dataflow and parameter/return detection.

Comment (jeffli678) [reply arch_x86.cpp:1945]: You may have a look at https://github.com/Vector35/binaryninja-api/commit/ce385dc73e2d264fc2404c75b060a8a42486e27b, which explains the purpose of system registers. Also, this PR should best be accompanied by some unit tests that verify the effectiveness of marking registers are system registers. I am not sure whether the general public can access our unit tests, so I can take over from here, i.e., add one or more commits to this PR and merge it later.

Comment (jeffli678) [reply arch_x86.cpp:1945]: And that you do not have to do more for this PR

**Pull request merged**